File: apps/java-sdk/src/main/java/com/firecrawl/models/ResearchModels.java
package com.firecrawl.models;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;
import java.util.Map;

public final class ResearchModels {
    private ResearchModels() {}

    public static class PaperResult {
        @JsonProperty("paperId")
        public String paperId;
        @JsonProperty("primaryId")
        public String primaryId;
        public Map<String, Object> ids;
        public String title;
        @JsonProperty("abstract")
        public String abstractText;
        public Double score;
        public Integer year;
        public List<String> authors;
        public String venue;
        public String url;
        public Map<String, Object> signals;
    }

    public static class PaperMetadata {
        @JsonProperty("paperId")
        public String paperId;
        public Map<String, Object> ids;
        public String title;
        @JsonProperty("abstract")
        public String abstractText;
        public String authors;
        public List<String> categories;
        @JsonProperty("createdDate")
        public String createdDate;
        @JsonProperty("updateDate")
        public String updateDate;
    }

    public static class Passage {
        public String text;
        public String section;
        public Integer page;
        public Double score;
        public Map<String, Object> metadata;
    }

    public static class SearchPapersResponse {
        public boolean success;
        public List<PaperResult> results;
    }

    public static class PaperMetadataResponse {
        public boolean success;
        public PaperMetadata paper;
    }

    public static class ReadPaperResponse {
        public boolean success;
        public PaperMetadata paper;
        @JsonProperty("paperId")
        public String paperId;
        public String query;
        public List<Passage> passages;
    }

    public static class SimilarPapersResponse {
        public boolean success;
        public List<PaperResult> results;
        @JsonProperty("poolSize")
        public Integer poolSize;
        public boolean truncated;
        public String note;
    }

    public static class GitHubSearchItem {
        public String resultType;
        public String repo;
        public String url;
        public String pageType;
        public Integer number;
        public Integer segmentCount;
        public String readmeUrl;
        public String title;
        public String snippet;
        public String contentMd;
        public Map<String, Object> scores;
    }

    public static class GitHubSearchResponse {
        public boolean success;
        public List<GitHubSearchItem> results;
    }

    public static class SearchPapersOptions {
        public Integer k;
        public List<String> authors;
        public List<String> categories;
        public String from;
        public String to;
    }

    public static class ReadPaperOptions {
        public Integer k;
    }

    public static class RelatedPapersOptions {
        public String mode;
        public Integer k;
        public Boolean rerank;
        public List<String> anchor;
    }

    public static class SearchGitHubOptions {
        public Integer k;
    }
}

File: apps/dot-net-sdk/Firecrawl/Models/ResearchModels.cs
using System.Text.Json.Serialization;

namespace Firecrawl.Models;

public class PaperResult
{
    [JsonPropertyName("paperId")]
    public string? PaperId { get; set; }

    [JsonPropertyName("primaryId")]
    public string? PrimaryId { get; set; }

    public Dictionary<string, object>? Ids { get; set; }
    public string? Title { get; set; }
    public string? Abstract { get; set; }
    public double? Score { get; set; }
    public int? Year { get; set; }
    public List<string>? Authors { get; set; }
    public string? Venue { get; set; }
    public string? Url { get; set; }
    public Dictionary<string, object>? Signals { get; set; }
}

public class PaperMetadata
{
    [JsonPropertyName("paperId")]
    public string? PaperId { get; set; }

    public Dictionary<string, object>? Ids { get; set; }
    public string? Title { get; set; }
    public string? Abstract { get; set; }
    public string? Authors { get; set; }
    public List<string>? Categories { get; set; }

    [JsonPropertyName("createdDate")]
    public string? CreatedDate { get; set; }

    [JsonPropertyName("updateDate")]
    public string? UpdateDate { get; set; }
}

public class Passage
{
    public string? Text { get; set; }
    public string? Section { get; set; }
    public int? Page { get; set; }
    public double? Score { get; set; }
    public Dictionary<string, object>? Metadata { get; set; }
}

public class SearchPapersResponse
{
    public bool Success { get; set; }
    public List<PaperResult>? Results { get; set; }
}

public class PaperMetadataResponse
{
    public bool Success { get; set; }
    public PaperMetadata? Paper { get; set; }
}

public class ReadPaperResponse
{
    public bool Success { get; set; }
    public PaperMetadata? Paper { get; set; }

    [JsonPropertyName("paperId")]
    public string? PaperId { get; set; }

    public string? Query { get; set; }
    public List<Passage>? Passages { get; set; }
}

public class SimilarPapersResponse
{
    public bool Success { get; set; }
    public List<PaperResult>? Results { get; set; }

    [JsonPropertyName("poolSize")]
    public int? PoolSize { get; set; }

    public bool Truncated { get; set; }
    public string? Note { get; set; }
}

public class GitHubSearchItem
{
    public string? ResultType { get; set; }

    public string? Repo { get; set; }
    public string? Url { get; set; }

    public string? PageType { get; set; }

    public int? Number { get; set; }

    public int? SegmentCount { get; set; }

    public string? ReadmeUrl { get; set; }

    public string? Title { get; set; }
    public string? Snippet { get; set; }

    public string? ContentMd { get; set; }

    public Dictionary<string, object>? Scores { get; set; }
}

public class GitHubSearchResponse
{
    public bool Success { get; set; }
    public List<GitHubSearchItem>? Results { get; set; }
}

public class SearchPapersOptions
{
    public int? K { get; set; }
    public List<string>? Authors { get; set; }
    public List<string>? Categories { get; set; }
    public string? From { get; set; }
    public string? To { get; set; }
}

public class ReadPaperOptions
{
    public int? K { get; set; }
}

public class RelatedPapersOptions
{
    public string? Mode { get; set; }
    public int? K { get; set; }
    public bool? Rerank { get; set; }
    public List<string>? Anchor { get; set; }
}

public class SearchGitHubOptions
{
    public int? K { get; set; }
}
